src/org/pentaho/reporting/platform/plugin/ParameterXmlContentHandler.java

```diff
@@ -738,18 +738,22 @@ Element createParameterElement( final ParameterDefinitionEntry parameter,
           final String namespace = namespaces[ i ];
           final String[] attributeNames = parameter.getParameterAttributeNames( namespace );
           for ( final String attributeName : attributeNames ) {
-            final String attributeValue = parameter.getParameterAttribute( namespace, attributeName, parameterContext );
-            // expecting: label, parameter-render-type, parameter-layout
-            // but others possible as well, so we set them all
-            final Element attributeElement = document.createElement( "attribute" ); // NON-NLS
-            attributeElement.setAttribute( "namespace", namespace ); // NON-NLS
-            attributeElement.setAttribute( "name", attributeName ); // NON-NLS
-            attributeElement.setAttribute( "value", attributeValue ); // NON-NLS
-
-            parameterElement.appendChild( attributeElement );
+            final String attributeValue =
```

Comment:
We need to skip some attributes at all - we don't need translate id on ui, we don't need post processing formula on ui (but it is exposed), we don't need... The boolean value is just to skip attribute with translate id. Is it ok to send to UI unused attributes? That could be rewritten with return string. But ParameterXmlContentHandler already so overload it will not help a lot in general picture.